tests/schemapack_/transform/test_base.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Test the base module."""
+
+
+import pytest
+from pydantic import ValidationError
+
+from metldata.schemapack_.builtin_transformations.null import NULL_TRANSFORMATION
+from metldata.schemapack_.transform.base import (
+    WorkflowDefinition,
+    WorkflowStep,
+)
+
+
+def test_workflow_definition_invalid_step_refs():
+    """Test that an invalid step reference raises an error."""
+    with pytest.raises(ValidationError):
```

Comment:
```suggestion
    with pytest.raises(
            ValidationError,
            match="Step non_existing_step referenced in step step2 is not defined"):
```

Reply:
See above, added to the refactoring list